### DISCUSSION TITLE
vtkdata as argument to click

### DISCUSSION THREAD

**Discussion by bigfooted** [Kitware/trame#296] [Q&A]
Hi,
I would like to save/download some contents of a vtk dataset by clicking a button.
So I have this data: 
`mb1 = vtkMultiBlockDataSet()`

and I want to give this data as an argument to a function export_files like this:

```
with vuetify.VBtn("download",click=(export_files,"[mb1]")):
        vuetify.VIcon("mdi-download-box-outline")
```

The function export_files is defined in an external module, and it grabs some data from the vtk dataset and saves it to a file. The above gives me "mb1 is not defined" and making it a state.mb1 does not work because its contents is not serializable. I am not familiar with this concept so don't really know how to serialize it. How can I make this work?

**Comment by jourdain** [comment]
Was [that example](https://github.com/Kitware/trame/blob/master/examples/validation/core/24_vtk_download_image.py) not matching what you aim to do?

Is your data on the server or on the client? If it is on the server, it does not need to pass the full data by the client (client is string expression "[mb1]"). Also you can use some kind of ID to let the server know which of its datasets needs to go through the call?

**Reply by bigfooted** [reply]
Hi, 
Yes, that example is actually pretty close to what I would like to do. But in my case, I would like to move the 'export_files(vtkmultiblock)' code to an external module. I just realized that I can simply first call `download_screenshot`, and just call the externally defined `export_files(vtkmultiblock)` from there.
And the data is all on the server.

**Reply by jourdain** [reply]
Indeed, all those calls can be done on the server side in plain Python.
